test/sql/aggregate_groupby_sql_test.cpp

```diff
@@ -0,0 +1,164 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// aggregate_groupby_sql_test.cpp
+//
+// Identification: test/sql/aggregate_groupby_sql_test.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <memory>
+
+#include "catalog/catalog.h"
+#include "common/harness.h"
+#include "concurrency/transaction_manager_factory.h"
+#include "optimizer/optimizer.h"
+#include "sql/testing_sql_util.h"
+#include "type/value_factory.h"
+
+namespace peloton {
+namespace test {
+
+class AggregateGroupBySQLTests : public PelotonTest {
+ protected:
+  void CreateAndLoadTable () {
+    // Create a table first
+    TestingSQLUtil::ExecuteSQLQuery(
+        "CREATE TABLE test(a INT PRIMARY KEY, b INT, c INT, d INT);");
+
+    // Insert tuples into table
+    TestingSQLUtil::ExecuteSQLQuery("INSERT INTO test VALUES (1, 2, 3, 1);");
+    TestingSQLUtil::ExecuteSQLQuery("INSERT INTO test VALUES (2, 2, 3, 1);");
+    TestingSQLUtil::ExecuteSQLQuery("INSERT INTO test VALUES (3, 2, 6, 1);");
+    TestingSQLUtil::ExecuteSQLQuery("INSERT INTO test VALUES (4, 2, 3, 2);");
+    TestingSQLUtil::ExecuteSQLQuery("INSERT INTO test VALUES (5, 3, 6, 2);");
+    TestingSQLUtil::ExecuteSQLQuery("INSERT INTO test VALUES (6, 4, 6, 2);");
+  }
+
+  void ExecuteSQLQueryAndCheckUnorderedResult (std::string query, std::unordered_set<std::string> ref_result) {
+    std::vector<StatementResult> result;
+    std::vector<FieldInfo> tuple_descriptor;
+    std::string error_message;
+    int rows_changed;
+
+    // Execute query
+    TestingSQLUtil::ExecuteSQLQuery(std::move(query), result, tuple_descriptor, rows_changed, error_message);
+    unsigned int rows = result.size() / tuple_descriptor.size();
+
+
+    // Build actual result as set of rows for comparison
+    std::unordered_set<std::string> actual_result(rows);
+    for (unsigned int i = 0; i < rows; i++) {
+      std::string row_string;
+
+      for (unsigned int j = 0; j < tuple_descriptor.size(); j++) {
+        row_string += TestingSQLUtil::GetResultValueAsString(result, i * tuple_descriptor.size() + j);
+        if (j < tuple_descriptor.size() - 1) {
+          row_string += "|";
+        }
+      }
+
+      actual_result.insert(std::move(row_string));
+    }
+
+
+    // Compare actual result to expected result
+    EXPECT_EQ(ref_result.size(), actual_result.size());
+
+    for (auto &result_row : ref_result) {
+      if (actual_result.count(result_row) == 0) {
+
+        EXPECT_EQ(ref_result, actual_result);
+        break;
+      }
+    }
+  }
+};
```

Comment:
This is a useful function that other tests can use. Can you move it to a utility class?

Reply:
I just talked with @pmenon and we decided to move the function as part of this PR. After this PR is merged I will adapt the test cases in #847 and #867 to use this utility function.